interfaces/apparmor/spec.go

```diff
@@ -181,7 +186,9 @@ func (spec *Specification) AddLayout(si *snap.Info) {
 			fmt.Fprintf(&buf, "  %s rw,\n", path)
 			WritableProfile(&buf, path, 2) // At least / and /some-top-level-directory
 		}
-		spec.AddUpdateNS(buf.String())
+		for _, line := range strings.SplitAfter(buf.String(), "\n") {
```

Comment:
I think this is problematic if we ever use this code for templated snippets because AppArmor allows multi-line rules for some types of rules, such as mount. For example the following is valid apparmor on my 19.04 Ubuntu AppArmor:
```
mount options=ro
/tmp/blah -> /tmp/blah2
```

If this type of apparmor rule is ever input into the de-duplication code you have here, then these two rules would become messed up/out of order IIUC:

```
mount options=ro
/tmp/blah -> /tmp/blah2,
```
and 
```
mount options=ro
/tmp/blah3 -> /tmp/blah4,
```
would get serialized out as
```
mount options=ro
/tmp/blah -> /tmp/blah2,
/tmp/blah3 -> /tmp/blah4,
```

which is obviously wrong and potentially very bad.

It's not clear to me yet if the code here is such that something else trying to write AppArmor could code accidentally cross paths with this and use the de-duplication. I will look more deeply to see if this is the case, but I think we ought to tread carefully to ensure that only snap-update-ns where the full profile is auto-generated uses this de-duplication and we can easily ensure no multi-line rules get input.

Reply:
We don't have any rules like that in this code so it's safe. I agree the split is unnecessary and I'm happy to remove it, it will blow up the patch size slightly since way more code will be affected, but we can remove it entirely in the end.

The annoying part of this pull request was the test changes to reflect the new state of the profiles we generate. We can certainly improve the details now.

Reply:
This is now changed to emit the lines directly. The buffer has been eliminated.